demos/monge_ampere_periodic.py

```diff
@@ -0,0 +1,174 @@
+#      65   Volume ratio  8.83   Variation (σ/μ) 3.49e-01   Residual 1.73e-03
+#      66   Volume ratio  8.80   Variation (σ/μ) 3.49e-01   Residual 1.29e-03
+#      67   Volume ratio  8.82   Variation (σ/μ) 3.49e-01   Residual 9.63e-04
+#      68   Volume ratio  8.80   Variation (σ/μ) 3.49e-01   Residual 6.87e-04
+#      69   Volume ratio  8.82   Variation (σ/μ) 3.49e-01   Residual 4.97e-04
+#      70   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 3.34e-04
+#      71   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 2.29e-04
+#      72   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.41e-04
+#      73   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 8.91e-05
+#      74   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 4.84e-05
+#      75   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 2.72e-05
+#      76   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.43e-05
+#      77   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.07e-05
+#      78   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 7.66e-06
+#      79   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 5.59e-06
+#      80   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 3.89e-06
+#      81   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 2.76e-06
+#      82   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.86e-06
+#      83   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.28e-06
+#      84   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 8.23e-07
+#      85   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 5.42e-07
+#      86   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 3.29e-07
+#      87   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 2.04e-07
+#      88   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 1.13e-07
+#      89   Volume ratio  8.81   Variation (σ/μ) 3.49e-01   Residual 6.46e-08
+#    Solver converged in 89 iterations.
+#
+# Again, plot the adapted mesh: ::
+
+import matplotlib.pyplot as plt
+from firedrake.pyplot import triplot
+
+fig, axes = plt.subplots()
+triplot(mover.mesh, axes=axes)
+axes.set_aspect(1)
+plt.savefig("monge_ampere_periodic-adapted_mesh.jpg")
+
+# .. figure:: monge_ampere_periodic-adapted_mesh.jpg
+#    :figwidth: 60%
+#    :align: center
+#
+# Observe that the outer boundary of the mesh is no longer square - each boundary
+# segment has been warped. However, you might be able to convice yourself that the warp
+# of the left boundary matches that of the right and that the warp of the top boundary
+# matches that of the bottom. To be absolutely certain, let's check that the area of the
+# mesh matches expectations. We can do this by simply integrating unity over the domain
+# associated with the adapted mesh. ::
+
+import numpy as np
+
+expected_area = 1.0
+assert np.isclose(assemble(Constant(1.0, domain=mover.mesh) * dx), expected_area)
```

Comment:
This raises a warning `/home/firedrake/firedrake/src/firedrake/firedrake/constant.py:73: FutureWarning: Giving Constants a domain is no longer supported. Instead please create a Function in the Real space.` I think this is equivalent to passing `domain` to the measure? This doesn't raise an error

```suggestion
assert np.isclose(assemble(Constant(1.0) * dx(domain=mover.mesh)), expected_area)
```

Reply:
Ah okay, I'd missed that. I noticed there's also `domain_constant` now, too.